crates/egui/src/containers/window.rs

```diff
@@ -1233,7 +1261,19 @@ impl TitleBar {
             let button_rect = button_rect.round_to_pixels(ui.pixels_per_point());
 
             ui.allocate_new_ui(UiBuilder::new().max_rect(button_rect), |ui| {
-                collapsing.show_default_button_with_size(ui, button_size);
+                if let Some(collapsed) = collapsed.as_ref() {
+                    if **collapsed != collapsing.is_collapsed() {
```

Comment:
"in case the user-provided has a different value than the internal 'is_collapsed', change the internal"